src/callable/primitive/sum.rs

```diff
@@ -0,0 +1,219 @@
+use r_derive::*;
+
+use crate::callable::core::*;
+use crate::error::*;
+use crate::lang::*;
+use crate::object::rep::Rep;
+use crate::object::*;
+
+#[derive(Debug, Clone, PartialEq)]
+#[builtin(sym = "sum")]
+pub struct PrimitiveSum;
+
+impl Callable for PrimitiveSum {
+    fn formals(&self) -> ExprList {
+        ExprList::from(vec![(None, Expr::Ellipsis(None))])
+    }
+
+    fn call(&self, args: ExprList, stack: &mut CallStack) -> EvalResult {
+        let (_, ellipsis) = self.match_arg_exprs(args, stack)?;
+
+        if ellipsis.is_empty() {
+            return EvalResult::Ok(Obj::Vector(Vector::from(Rep::from(vec![0.0]))));
+        }
+
+        let objects: Vec<Obj> = force_closures(ellipsis, stack)
+            .unwrap()
+            .into_iter()
+            .map(|(_, value)| value)
+            .collect();
+
+        let mut any_numeric: bool = false;
+
+        for obj in &objects {
+            match obj {
+                Obj::Vector(Vector::Numeric(..)) => {
+                    any_numeric = true;
+                    break;
+                }
+                Obj::Vector(Vector::Logical(..)) | Obj::Vector(Vector::Integer(..)) => {
```

Comment:
I would start with being on the more conservative side here and in general try to avoid `NULL`s (aka the One Billion Dollar mistake) as much as possible. Not allowing `NULL`s in the sum function should only be a problem when it can happen often that one accidentally passes objects that are `NULL`, but that we expect to be numeric / integers / logicals. As I think these type of event should also be avoided, I think not handling `NULL` should pose no incovenience to users and just be a safeguard from bugs.

Reply:
I would even argue that `c()` should maybe throw an error instead of returning `NULL`, what do you think?

Reply:
Yeah, I agree. I'd prefer to go the route of expecting more consistent input types as a general rule throughout the base library. Let's keep it as is, and we can always come back to it if we decide differently.